app/server/appsmith-server/src/main/java/com/appsmith/server/solutions/ImportExportApplicationService.java

```diff
@@ -0,0 +1,576 @@
+import com.appsmith.server.services.NewPageService;
+import com.appsmith.server.services.OrganizationService;
+import com.appsmith.server.services.SequenceService;
+import com.appsmith.server.services.SessionUserService;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.bson.types.ObjectId;
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferUtils;
+import org.springframework.http.MediaType;
+import org.springframework.http.codec.multipart.Part;
+import org.springframework.stereotype.Component;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.lang.reflect.Type;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class ImportExportApplicationService {
+    private final DatasourceService datasourceService;
+    private final SessionUserService sessionUserService;
+    private final NewActionRepository newActionRepository;
+    private final DatasourceRepository datasourceRepository;
+    private final PluginRepository pluginRepository;
+    private final OrganizationService organizationService;
+    private final ApplicationService applicationService;
+    private final NewPageService newPageService;
+    private final ApplicationPageService applicationPageService;
+    private final NewPageRepository newPageRepository;
+    private final NewActionService newActionService;
+    private final SequenceService sequenceService;
+    private final ExamplesOrganizationCloner examplesOrganizationCloner;
+
+    private static final Set<MediaType> ALLOWED_CONTENT_TYPES = Set.of(MediaType.APPLICATION_JSON);
+
+    public Mono<ApplicationJson> exportApplicationById(String applicationId) {
+        ApplicationJson applicationJson = new ApplicationJson();
+        Map<String, String> pluginMap = new HashMap<>();
+        Map<String, String> datasourceMap = new HashMap<>();
+        Map<String, String> newPageIdMap = new HashMap<>();
+    
+        if (applicationId == null || applicationId.isEmpty()) {
+            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, FieldName.APPLICATION_ID));
+        }
+
+        return pluginRepository
+            .findAll()
+            .collectList()
+            .map(pluginList -> {
+                pluginList.forEach(plugin -> pluginMap.put(plugin.getId(), plugin.getPackageName()));
+                return pluginList;
+            })
+            .flatMap(ignore -> applicationService.findById(applicationId, AclPermission.READ_APPLICATIONS))
+            .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.NO_RESOURCE_FOUND, FieldName.APPLICATION, applicationId)))
```

Comment:
Let's use the ACL error message since, the resource might be present, but the user might not have permissions for it.

```suggestion
            .switchIfEmpty(Mono.error(new AppsmithException(AppsmithError.ACL_NO_RESOURCE_FOUND, FieldName.APPLICATION, applicationId)))
```

Reply:
Updated code pushed!